packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTFontUtils.mm

```diff
@@ -166,6 +166,13 @@ static RCTFontStyle RCTGetFontStyle(UIFont *font)
   } else {
     NSArray<NSString *> *fontNames = [UIFont fontNamesForFamilyName:fontProperties.family];
 
+    if (fontNames.count == 0) {
+      font = [UIFont fontWithName:fontProperties.family size:effectiveFontSize];
+      if (font) {
+        fontNames = [UIFont fontNamesForFamilyName:font.familyName];
+      }
+    }
+
```

Comment:
This logic is very similar to what's immediately below - should it be restructured more? 

```
fontNames = [...]
if (fontNames.count == 0) {
  font = [UIFont fontWithName:fontProperties.family size:effectiveFontSize];
  if (font) {
    fontNames = [UIFont fontNamesForFamilyName:font.familyName];
  } else {
    // Fallback to system font.
    font = [UIFont systemFontOfSize:effectiveFontSize weight:fontProperties.weight];
}

if (fontNames.count >  1) {}